packages/verification-core/src/pdf/PdfService.js

```diff
@@ -121,10 +121,55 @@ export default class PdfService {
                 })
 
                 worker.postMessage({
+                    cmd: 'extract_encryption_keys',
                     pdfWasmModule: this.pdfWasmModule,
                     pdfBytes
                 })
             }
         })
     }
+
+    /**
+     * Decrypt the given pdf bytes with the given encryption key
+     *
+     * @param {Uint8Array} pdfBytes
+     * @param {string} encryptionKey
+     *
+     * @returns {Promise<void>}
+     */
+    async decryptPdf(pdfBytes, encryptionKey) {
+        await this.waitUntilLoaded()
+
+        let decryptedPdfBytes = null
+        if (this.isNonChromiumEdge === true) {
+            decryptedPdfBytes = await PdfWasmWrapper.decryptPdf(pdfBytes, encryptionKey)
+        }
+
+        return new Promise((resolve, reject) => {
+            if (this.isNonChromiumEdge === true) {
+                if (decryptedPdfBytes === null) {
+                    reject(new Error('decryptedPdfBytes is null'))
+                }
+
+                resolve(decryptedPdfBytes)
+            } else {
```

Comment:
I'd remove the `else`-brackets since in the `if` case we either reject or resolve which should return either way.

Reply:
The if-else is needed for handling it differently if it isn't a chromium browser.

Reply:
my proposal was more an aesthetic one to simplify the if, else structure and remove the `else` , as far as i know it behaves the same because reject and resolve both return:
```
return new Promise((resolve, reject) => {
            if (this.isNonChromiumEdge === true) {
                if (decryptedPdfBytes === null) {
                    reject(new Error('decryptedPdfBytes is null'))
                }

                resolve(decryptedPdfBytes)
            }
            const worker = new PdfWorker()

            worker.addEventListener('message', function(e) {
                if (e.data.status === true) {
                    resolve(e.data.decryptedPdfBytes)
                } else {
                    reject(e.data.error)
                }
            }, false)

            worker.postMessage({
                cmd: 'decrypt_pdf',
                pdfWasmModule: this.pdfWasmModule,
                pdfBytes,
                encryptionKey
            })
        })
```

Reply:
Ah sorry i misunderstood you. The resolve or reject wouldn't stop the execution of the code below so the else is necessary here. Had this issue a while ago.

Reply:
good to know, thanks for clarifying